wolfcrypt/src/rsa.c

```diff
@@ -1526,16 +1545,20 @@ static int RsaUnPad_OAEP(byte *pkcsBlock, unsigned int pkcsBlockLen,
         return BAD_FUNC_ARG;
     }
 
+#if defined(WOLFSSL_SMALL_STACK) && !defined(WOLFSSL_NO_MALLOC)
     tmp = (byte*)XMALLOC(pkcsBlockLen, heap, DYNAMIC_TYPE_RSA_BUFFER);
```

Comment:
Should you also add a check to make sure `pkcsBlockLen < RSA_MAX_SIZE/8 + RSA_PSS_PAD_SZ`?

Reply:
Previous line `(pkcsBlockLen < (2 * (word32)hLen + 2))` check covers it and would fail before this line.

```
min when hLen = 0: pkcsBlockLen < 2
max when hLen=64: pkcsBlockLen < 130
```
```
pkcsBlockLen < RSA_MAX_SIZE/8 + RSA_PSS_PAD_SZ
pkcsBlockLen < 520
```

Reply:
Doesn't that mean you are not properly sizing the `tmp` buffer and it could be smaller?

Reply:
@dgarske, you are correct that _tmp_ size could be smaller.  AFACT, technically, _tmp_ size can be RSA_MAX_SIZE/8  - (hLen + 1). So it can be made smaller by 1 byte? seems better for readability to not subtract anything. But it is your call.